test/extended/images/mongodb_ephemeral.go

```diff
@@ -0,0 +1,71 @@
+package images
+
+import (
+	"fmt"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/db"
+	"time"
+)
+
+var _ = g.Describe("images: mongodb: ephemeral template", func() {
+	defer g.GinkgoRecover()
+
+	templatePath := exutil.FixturePath("..", "..", "examples", "db-templates", "mongodb-ephemeral-template.json")
+	oc := exutil.NewCLI("mongodb-create", exutil.KubeConfigPath()).Verbose()
+
+	g.Describe("creating from a template", func() {
+		g.It(fmt.Sprintf("should process and create the %q template", templatePath), func() {
+
+			g.By("creating a new app")
+			o.Expect(oc.Run("new-app").Args("-f", templatePath).Execute()).Should(o.Succeed())
+
+			g.By("expecting the mongodb service get endpoints")
+			o.Expect(oc.KubeFramework().WaitForAnEndpoint("mongodb")).Should(o.Succeed())
+
+			g.By("expecting the mongodb pod is running")
+			podNames, err := exutil.WaitForPods(
+				oc.KubeREST().Pods(oc.Namespace()),
+				exutil.ParseLabelsOrDie("name=mongodb"),
+				exutil.CheckPodIsRunningFn,
+				1,
+				1*time.Minute,
+			)
+			o.Expect(err).ShouldNot(o.HaveOccurred())
+			o.Expect(podNames).Should(o.HaveLen(1))
+
+			g.By("expecting the mongodb service is answering for ping")
+			mongo := db.NewMongoDB(podNames[0], "")
+
+			for times := 0; times < 10; times++ {
```

Comment:
Just to document the idea in a public place...

This type of loop could be replaced with code that blocks until something that we're waiting for is "ready", without recurring to `time.Sleep`.

In this case here, we're waiting for a deployment of MongoDB to be ready.


We can use `oc logs -f dc/NAME` to block until a deployment of `NAME` is finished, e.g.:

```
[vagrant@localhost origin]$ oc deploy --latest beego-example
Started deployment #5
[vagrant@localhost origin]$ oc logs -f dc/beego-example
I0126 16:46:39.923942       1 deployer.go:198] Deploying from demo/beego-example-4 to demo/beego-example-5 (replicas: 1)
I0126 16:46:39.933468       1 recreate.go:93] Scaling demo/beego-example-4 down to zero
I0126 16:46:41.996868       1 recreate.go:118] Scaling demo/beego-example-5 to 1
I0126 16:46:44.114626       1 recreate.go:136] Deployment beego-example-5 successfully made active
```

We don't need the actual logs, they can be sent to `/dev/null`, and we can have a single call to `mongo.isReady` to ensure the deployment was successful.